ecmascript/parser/src/parser/stmt/module_item.rs

```diff
@@ -177,18 +177,90 @@ impl<'a, I: Tokens> Parser<I> {
     }
 
     /// Parse `foo`, `foo2 as bar` in `import { foo, foo2 as bar }`
-    fn parse_import_specifier(&mut self) -> PResult<ImportSpecifier> {
+    fn parse_import_specifier(&mut self, is_type_only_import: bool) -> PResult<ImportSpecifier> {
         let start = cur_pos!(self);
         match cur!(self, false) {
             Ok(&Word(..)) => {
-                let orig_name = self.parse_ident_name()?;
+                let mut is_type_only = false;
+                let mut orig_name = self.parse_ident_name()?;
+
+                // Handle:
+                // `import { type xx } from 'mod'`
+                // `import { type xx as yy } from 'mod'`
+                // `import { type as } from 'mod'`
+                // `import { type as as } from 'mod'`
+                // `import { type as as as } from 'mod'`
+                if orig_name.sym == js_word!("type")
+                    && is!(self, IdentName)
+                    && self.syntax().typescript()
+                    // invalid: `import type { type something } from 'mod'`
```

Comment:
I think we should still parse this scenario even though it is invalid. It would be useful in dprint-plugin-typescript for example to parse this then convert `import type { type something} from 'mod'` to `import type { something } from "mod";`. Thoughts? (by the way, nice PR)

Reply:
I think dprint is a formatter, and it shoudn't accept code with invalid syntax.

Reply:
I prefer that this would be syntax errors, but I'm OK to treat it as valid syntax, because this isn't a huge change.

> The biggest annoying problem is that once I change the code, rustc will cost very very long time.

Reply:
It can be syntax error and can be valid ast in same time, and I think it's the right direction to go.
`emit_err` can be used for such purpose.